setup/main.go

```diff
@@ -0,0 +1,1653 @@
+		runPgSleep,
+	}
+
+	var setupState SetupState
+	var verbose bool
+	var logFile string
+	flag.StringVar(&setupState.ConfigFilename, "config", defaultConfigFile, "Specify alternative path for config file")
+	flag.BoolVar(&verbose, "verbose", false, "Include verbose logging output")
+	flag.StringVar(&logFile, "log", "", "Save output to log file (always includes verbose output)")
+	flag.Parse()
+	var logger Logger
+	if logFile != "" {
+		logger = NewLogger()
+		log, err := os.Create(logFile)
+		if err != nil {
+			fmt.Errorf("could not open %s for writes: %s", logFile, err)
+			return
+		}
+		defer log.Close()
+		outputBoth := io.MultiWriter(os.Stdout, log)
+		logger.StandardOutput = outputBoth
+		if verbose {
+			logger.VerboseOutput = outputBoth
+		} else {
+			logger.VerboseOutput = log
+		}
+	} else {
+		logger = NewLogger()
+		if verbose {
+			logger.VerboseOutput = os.Stdout
+		}
+	}
+	setupState.Logger = &logger
+
+	// TODO: check for root?
+
+	setupState.Log(`Welcome to the pganalyze collector installer!
+
+We will go through a series of steps to set up the collector to monitor your
+Postgres database. We will not make any changes to your database or system
+without confirmation.
+
+At a high level, we will:
+
+ 1. Configure a database user and helper functions for the collector, with minimal access
+ 2. Update the collector configuration file with these settings
+ 3. Set up the pg_stat_statements extension in your database for basic query performance monitoring
+ 4. (Optional) Make log-related configuration settings changes to enable our Log Insights feature
+ 5. (Optional) Set up EXPLAIN plan collection to enable our Automated EXPLAIN feature
+
+At each step, we'll check if any changes are necessary, and if so, prompt you to
+provide input or confirm any required changes.
+
+Changes to Postgres configuration settings will be done with the ALTER SYSTEM command.
+If you later need to refine any of these, make sure to use ALTER SYSTEM or ALTER SYSTEM RESET,
+since otherwise, the ALTER SYSTEM changes will override any direct config file edits. Learn
+more at https://www.postgresql.org/docs/current/sql-altersystem.html .
+
+You can stop at any time by pressing Ctrl+C.
+
+If you stop before completing setup, you can resume by running the installer
+again. We can pick up where you left off.`)
+	setupState.Log("")
+	var doSetup bool
+	err := survey.AskOne(&survey.Confirm{
+		Message: "Continue with setup?",
+		Default: false,
+	}, &doSetup)
+	if err != nil {
+		setupState.Log("  automated setup failed: %s", err)
+	}
+	if !doSetup {
+		setupState.Log("Exiting...")
+		os.Exit(0)
+	}
+
+	for _, step := range steps {
+		err := doStep(&setupState, step)
+		if err != nil {
+			os.Exit(1)
+		}
+	}
+}
+
+func doStep(setupState *SetupState, step *Step) error {
+	if step.Check == nil {
+		panic("step missing completion check")
+	}
+	setupState.Logger.StartStep(step.Description)
+	defer setupState.Logger.EndStep()
+	done, err := step.Check(setupState)
```

Comment:
When you're running this as non-root, it will currently fail like this:

```
? Continue with setup? Yes
  ✗ failed to check status: open /etc/pganalyze-collector.conf: permission denied
```

It might be good to do an explicit permission check, and say that you should re-run with sudo if the user is not root?

Reply:
Yeah, makes sense. Just check `os.Geteuid`, right?

Reply:
Yeah, I think so